### PR TITLE
[Merged by Bors] - refactor(*): change definition of `Set.image2` etc

### PR DESCRIPTION
**Pull request by urkud**
- Redefine `Set.image2` to use `∃ a ∈ s, ∃ b ∈ t, f a b = c` instead of `∃ a b, a ∈ s ∧ b ∈ t ∧ f a b = c`.
- Redefine `Set.seq` as `Set.image2`. The new definition is equal to the old one but `rw [Set.seq]` gives a different result.
- Redefine `Filter.map₂` to use `∃ u ∈ f, ∃ v ∈ g, image2 m u v ⊆ s` instead of `∃ u v, u ∈ f ∧ v ∈ g ∧ ...`
- Update lemmas like `Set.mem_image2`, `Finset.mem_image₂`, `Set.mem_mul`, `Finset.mem_div` etc

The two reasons to make the change are:

- `∃ a ∈ s, ∃ b ∈ t, _` is a `simp`-normal form, and
- it looks a bit nicer.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Lattice.lean [hunk 1]
```diff
@@ -2081,8 +2081,7 @@ section Seq
 
 /-- Given a set `s` of functions `α → β` and `t : Set α`, `seq s t` is the union of `f '' t` over
 all `f ∈ s`. -/
-def seq (s : Set (α → β)) (t : Set α) : Set β :=
-  { b | ∃ f ∈ s, ∃ a ∈ t, (f : α → β) a = b }
+def seq (s : Set (α → β)) (t : Set α) : Set β := image2 (fun f a ↦ f a) s t
```
Comment (YaelDillies) [inline Mathlib/Data/Set/Lattice.lean:2084]: Do you think this is better?
```suggestion
def seq (s : Set (α → β)) (t : Set α) : Set β := image2 id s t
```

Comment (eric-wieser) [reply Mathlib/Data/Set/Lattice.lean:2084]: `Function.eval` would probably be clearest.

Comment (YaelDillies) [reply Mathlib/Data/Set/Lattice.lean:2084]: Yep, that's backward.

Comment (urkud) [reply Mathlib/Data/Set/Lattice.lean:2084]: I reverted this commit because it generates `id` here and there after `simp only [seq]`.

Comment (YaelDillies) [reply Mathlib/Data/Set/Lattice.lean:2084]: What about
```suggestion
def seq (s : Set (α → β)) (t : Set α) : Set β := image2 (fun f ↦ f) s t
```
then?

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**